### PR TITLE
Update for XML Parsing

### LINKED ISSUE #2: Implement New Orphanet Parser Feature
The project involves creating a new parser to effectively manage and integrate the comprehensive rare diseases dataset from Orphanet, which is available on their Orphanet Scientific Knowledge Files website. This data is rich in content, including rare disease alignments with various medical classifications like SNOMED CT, ICD-10, ICD-11, OMIM, UMLS, MeSH, MedDRA, and GARD. In addition, it provides detailed information on classifications of rare diseases, their linearisation, associated genes, clinical signs and symptoms, functional consequences, epidemiology, and natural history. We aim to develop a parser to handle these diverse data types and bring them into a unified format that fits our system's needs. This will significantly enhance our capacity to utilize the Orphanet dataset for advanced analysis and application in rare disease studies.

Task Checklist:

- [ ]  Analyze Orphanet's data format and structure to understand integration requirements.
- [ ]  Design the parser architecture, ensuring compatibility with our current system.
- [ ]  Develop the parser code, incorporating necessary features for Orphanet data handling.
- [ ]  Test the parser with sample Orphanet data to validate functionality.
- [ ]  Integrate the parser into the existing system, ensuring seamless data flow.
- [ ]  Conduct thorough unit and integration testing to ensure stability and performance.
- [ ]  Please document the new feature comprehensively and update user manuals accordingly.

The Link:
https://www.orphadata.com/orphanet-scientific-knowledge-files/

### PR DESCRIPTION
**Pull request by fbkaragoz**
#2 issue has been reviewed, and more adjusted folder classifications have been proposed. We will also adhere to classes for dict-type formats to be more minimalized in the script. 

I will also apply some valuable information to further the data we are parsing. One such example is given in metrics. Sh in our repo.

successful merge of this pull request closes #2

### CODE REVIEW

File: prometheus_test/.DS_Store
Comment (themkdemiiir) [inline prometheus_test/.DS_Store:1]: What is this file?

**Review by themkdemiiir — CHANGES_REQUESTED** [review]
do not push data files into a branch

File: prometheus_test/orphanet_data/orphanet_data_csv/combined_diseases.csv
Comment (themkdemiiir) [inline prometheus_test/orphanet_data/orphanet_data_csv/combined_diseases.csv:1]: do not push data files into a branch

File: prometheus_test/orphanet_data/orphanet_data_csv/en_product3_181.csv
Comment (themkdemiiir) [inline prometheus_test/orphanet_data/orphanet_data_csv/en_product3_181.csv:1]: do not push data files into a branch

File: prometheus_test/orphanet_data/orphanet_xml/en_product3_181.xml
Comment (themkdemiiir) [inline prometheus_test/orphanet_data/orphanet_xml/en_product3_181.xml:1]: do not push data files into a branch

File: prometheus_test/sh_scripts/metrics.sh
Comment (themkdemiiir) [inline prometheus_test/sh_scripts/metrics.sh:1]: do not push sh_scripts into the branch

File: prometheus_test/py_scripts/test_main.py
Comment (themkdemiiir) [inline prometheus_test/py_scripts/test_main.py:1]: this is a parser main file should be named orphanet/orphanet_parser.py
tests should be pushed in this convention tests/orphanet_tests/test_orphanet_parser.py

**Review by themkdemiiir — CHANGES_REQUESTED** [review]
Review your code use functional programming where necessary

File: prometheus_test/orphanet_data/orphanet_xml/en_product3_181.xml
Comment (rekurrenzk) [reply prometheus_test/orphanet_data/orphanet_xml/en_product3_181.xml:1]: Branches are created. I am using the zk branch to parse the script right now. I am issuing a pr to merge the latest update.

File: prometheus_test/py_scripts/test_main.py
Comment (rekurrenzk) [reply prometheus_test/py_scripts/test_main.py:1]: ok

**Comment by rekurrenzk** [comment]
@cdliproject/cdli-core-team all problems are resolved. Please review the latest update asap, and let me know what you think.

File: sh_scripts/metrics.sh
Comment (themkdemiiir) [inline sh_scripts/metrics.sh:1]: put this file into the .gitignore

File: py_scripts/parser.py
Comment (themkdemiiir) [inline py_scripts/parser.py:1]: review the code make it class oriented, use data classes, do not use pandas

File: py_scripts/parser.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            
+            'ClassificationID': self.classification_id,
+            'DisorderID': self.disorder_id,
+            'DisorderType' : self.disorder_type,
+            'OrphaCode': self.orpha_code,
+            'Name': self.name,
+            'ExpertLink': self.expert_link,
+            'MetaData': self.meta_data
+            
+        }
```
Comment (themkdemiiir) [inline py_scripts/parser.py:64]: If a class is required for a certain task, it is recommended to use data classes.

File: py_scripts/parser.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        csv_file_path = os.path.join(directory, csv_file_name)
+        
+        diseases_df.to_csv(csv_file_path, index=False)
+        print(f'File saved to {csv_file_path}')
+
+
+def main():
+    os.makedirs(directory, exist_ok=True)
+    xml_files = get_xml_files(raw_data_path)
+    process_xml_files(xml_files, directory)
```
Comment (themkdemiiir) [inline py_scripts/parser.py:197]: look good but use python types line  def func(x: str) -> str:
etc

File: lib/exceptions.py
Comment (themkdemiiir) [inline lib/exceptions.py:1]: why did you delete this files?

File: orphanet_data/orphanet_csv/en_product3_181.csv
Comment (themkdemiiir) [inline orphanet_data/orphanet_csv/en_product3_181.csv:1]: Do not push datafiles into repo

**Review by themkdemiiir — CHANGES_REQUESTED** [review]
Write tests for your data under tests/test_orphanet_parser.py
use unittest

File: sh_scripts/metrics.sh
Comment (rekurrenzk) [reply sh_scripts/metrics.sh:1]: We should keep it in a separate folder w/o embedding it into .gitignore. I just wanted to let you know that further deployments will be facilitated from sh scripts. I see no point in removing it from viable visibility.

File: lib/exceptions.py
Comment (rekurrenzk) [reply lib/exceptions.py:1]: the branch zk is designed to store only parser modification. you can initiate pr for other scripts required into the main

File: py_scripts/parser.py (hunk 2 above)
Comment (rekurrenzk) [reply py_scripts/parser.py]: Please check out the recent pr I am initiating, and return any insight if the structure is still needed.

File: py_scripts/parser.py (hunk 1 above)
Comment (rekurrenzk) [reply py_scripts/parser.py]: For current main.py, I have only used functions. Please review the latest pr.

**Comment by rekurrenzk** [comment]
> Write tests for your data under tests/test_orphanet_parser.py use unittest

@Simurgan can have better skills for deploying tests for orphanet. Help needed.

**Comment by rekurrenzk** [comment]
@themkdemiiir please review the changes ASAP.
